src/physics.F90

```diff
@@ -459,11 +470,17 @@ subroutine elastic_scatter(i_nuclide, rxn, E, uvw, mu_lab, wgt)
     vel = sqrt(E)
 
     ! compute cosine of scattering angle in LAB frame by taking dot product of
-    ! neutron's pre- and post-collision angle
-    mu_lab = dot_product(uvw, v_n) / vel
+    ! neutron's pre- and post-collision unit vectors
+    if (iso_lab) then
+      uvw(1) = TWO * prn() - ONE
+      phi = TWO * PI * prn()
+      uvw(2) = cos(phi) * sqrt(ONE - uvw(1)*uvw(1))
+      uvw(3) = sin(phi) * sqrt(ONE - uvw(1)*uvw(1))
```

Comment:
It seems like the logic for isotropic scattering ought to be at the end of `subroutine scatter` rather than inside each of `elastic_scatter`, `sab_scatter`, and `inelastic_scatter`. If we did that, those three routines wouldn't need any modifications (I think?). Also, we can move the calculation of mu_lab to the end of `scatter` as well.